src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyState, action.extraData);
+        break;
+
+      case 'clear':
+        copyState = {};
+        break;
+
+      default:
```

Comment:
No no, it's a critical mistake here. For example, if I use not defined action it will try to remove my fields, it's bad)
I think you a little bit misunderstand this comment:
https://github.com/mate-academy/js_stateful-clones/pull/1970#discussion_r1101334568

Here was meaning next:
```
      default:
        throw new Error('Unexpected action');
```

Please return back your previous case and add code above ^ for default case.